specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-03-01/managedClusters.json

```diff
@@ -4292,6 +4504,81 @@
         }
       }
     },
+    "ExtensionAddon": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "$ref": "#/definitions/SystemData",
+          "readOnly": true,
+          "description": "The system metadata related to this resource."
+        },
+        "properties": {
+          "description": "Properties of a extension add-on.",
+          "$ref": "#/definitions/ExtensionAddonProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "Extension add-on."
+    },
+    "ExtensionAddonProperties": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The current provisioning state of extension add-on.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Failed",
+            "Succeeded"
+          ],
+          "x-ms-enum": {
+            "name": "ExtensionAddonProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "addonName": {
+          "type": "string",
+          "description": "Corresponding name if the add-on is also embedded in AKS."
+        },
+        "enableManagedIdentity": {
+          "type": "boolean",
+          "description": "Whether managed identity is required for the extension add-on or not."
+        },
+        "identity": {
```

Comment:
identity should be a top level property at the level of id, name type etc. properties. Why is it nested inside properties?
https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingUserAssigned.html?q=user%20ass

Reply:
@ravbhatnagar Thanks for reviewing. The intention of the "identity" here is different to the identity mentioned in the linked document. This identity is created by AKS and user PUT/PATCH operation is not allowed. The scope of this field is more narrow, like a property of the extension addon.

Reply:
If its not the top level identity property it must be named something different.  However, the purpose of this property seems odd.  Why is AKS creating a user assigned identity for the user behind the scenes?  What subscription does that get created in?  Why can the user not create it and assign it to the extensionAddon resource like normal user assigned identities?

Reply:
This API is private to Azure, only accessible by the Azure Arc RP. So, user cannot create an identity and assign it. AKS creates identity in the node resource group for each extensionAddon, then Arc RP reads the identity info. Arc RP will then be able to authorize the requests from workloads on AKS cluster based on the identity in request token.
I'm not sure if private API is supposed to be added to this repo. If not, I can give up this change.
I'm OK with a different name of this property if "identity" is considered a reserved name, although I think it's not ambiguous since it's in the "properties".

Reply:
yes, internal APIs are not documented in the swagger repo.  Please ensure this proposed design goes through some level of security review though.

Reply:
@pilor Thanks! Will do that.